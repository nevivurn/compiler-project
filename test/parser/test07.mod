//
// test07
//
// parser test:
// - char data types
//

module test07;

var c: char;

begin
  c := '1';
  c := '\n';
  c := '\x41'
end test07.
