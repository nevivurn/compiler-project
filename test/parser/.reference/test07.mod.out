parsing 'test07.mod'...
successfully parsed.
  AST:
    CAstScope: 'test07'
      symbol table:
        [[
          [ *DIM(<ptr(8) to <NULL>>,<integer>) --> <integer>           ]
          [ *DOFS(<ptr(8) to <NULL>>) --> <integer>           ]
          [ *ReadInt() --> <integer>           ]
          [ *ReadLong() --> <longint>           ]
          [ *WriteChar(<char>) --> <NULL>           ]
          [ *WriteInt(<integer>) --> <NULL>           ]
          [ *WriteLn() --> <NULL>           ]
          [ *WriteLong(<longint>) --> <NULL>           ]
          [ *WriteStr(<ptr(8) to array of <char>>>) --> <NULL>           ]
          [ @c        <char>           c           ]
          [ main     <NULL>           ]
        ]]
      statement list:
        := <char>
          [ @c        <char>           c ] <char>
          49 <char>
        := <char>
          [ @c        <char>           c ] <char>
          10 <char>
        := <char>
          [ @c        <char>           c ] <char>
          65 <char>
      nested scopes:
        empty.
    


run the following command to convert the .dot file into a PDF:
  dot -Tpdf -otest07.mod.ast.pdf test07.mod.ast.dot


Done.
